### ISSUE TITLE
Table-valued or composite types parameters for rpc functions

### ISSUE THREAD

**Issue by Barrokgl** [PostgREST/postgrest#3595]
* PostgreSQL version: image postgres:16-alpine
* PostgREST version: image postgrest/postgrest:latest
* Operating system: Mac OS Ventura 13.4

### Description of issue

Does PostgREST rpc functions support table-valued or composite types parameters?

example code with composite type parameter: 
`CREATE TYPE api.my_param_type AS (
    "name" VARCHAR(255)
);

CREATE OR REPLACE FUNCTION api.example_function(my_param api.my_param_type) RETURNS void AS $$ 
BEGIN

END;
$$ LANGUAGE plpgsql;`

as a result for composite types approach, parameters mapped as strings, which disallows function call with nested json
<img width="765" alt="Screenshot 2024-06-15 at 15 00 11" src="https://github.com/PostgREST/postgrest/assets/20322348/ff4fab8e-8bce-4030-adb5-7e5afca256db">

but works with string which contains columns separated by commas:
<img width="654" alt="Screenshot 2024-06-15 at 15 28 41" src="https://github.com/PostgREST/postgrest/assets/20322348/6a56bdfb-5077-4ac4-aa1e-304db1b5d19f">

example code with table-valued parameter: 
`CREATE TABLE internal.company (
    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
    "name" VARCHAR(255) NOT NULL,
    "description" TEXT,
    created_at TIMESTAMP NOT NULL DEFAULT NOW(),
    updated_at TIMESTAMP NOT NULL DEFAULT NOW()
);

CREATE VIEW api.company AS
SELECT
    id,
    "name",
    description,
    created_at,
    updated_at
FROM
    internal.company;
    
CREATE OR REPLACE FUNCTION api.example_function_2(my_param api.company) RETURNS void AS $$ 
BEGIN`

same behavoir with table-valued parameter:
<img width="1280" alt="Screenshot 2024-06-15 at 15 32 03" src="https://github.com/PostgREST/postgrest/assets/20322348/2353bcf5-f8f0-4f59-814a-401cfeb71e79">

is it even documented behaviour (could not found anything in the docs) and is there is way to pass these parameters values as json objects?
thanks

**Comment by Barrokgl** [comment]
> > is there is way to pass these parameters values as json objects?
> 
> Are you getting any error when you do so?

@wolfgangwalther yes, it works fine when calling directly there is no errors, but when relying on openapi schema, like with swagger ui or generated client.
i'm getting error because in the schema parameter type is defined as string:
<img width="636" alt="Screenshot 2024-06-15 at 16 30 40" src="https://github.com/PostgREST/postgrest/assets/20322348/ef0b92ca-76f6-4232-b94a-02aed851a353">

generated openapi schema for function with compisite types:
```
{
  "/rpc/example_function": {
    "get": {
      "tags": [
        "(rpc) example_function"
      ],
      "produces": [
        "application/json",
        "application/vnd.pgrst.object+json;nulls=stripped",
        "application/vnd.pgrst.object+json"
      ],
      "parameters": [
        {
          "format": "api.my_param_type",
          "in": "query",
          "name": "my_param",
          "required": true,
          "type": "string"
        }
      ],
      "responses": {
        "200": {
          "description": "OK"
        }
      }
    },
    "post": {
      "tags": [
        "(rpc) example_function"
      ],
      "produces": [
        "application/json",
        "application/vnd.pgrst.object+json;nulls=stripped",
        "application/vnd.pgrst.object+json"
      ],
      "parameters": [
        {
          "in": "body",
          "name": "args",
          "required": true,
          "schema": {
            "properties": {
              "my_param": {
                "format": "api.my_param_type",
                "type": "string"
              }
            },
            "required": [
              "my_param"
            ],
            "type": "object"
          }
        },
        {
          "$ref": "#/parameters/preferParams"
        }
      ],
      "responses": {
        "200": {
          "description": "OK"
        }
      }
    }
  }
}
```

**Comment by Barrokgl** [comment]
@wolfgangwalther okay, thank you!

**Comment by tjmcdonough** [comment]
Is it possible to return the actual response of the SQL function instead of the below always getting returned?

```
 "responses": {
    "200": {
      "description": "OK"
    }
}
```
